### PR TITLE
[Improvement] Add element and elementservice resolvers

### PR DESCRIPTION
**Pull request by mattamon**
Adding more methods for resolving specifc element types and services

### CODE REVIEW

File: src/Models/Asset/AssetResolver.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+namespace Pimcore\Bundle\StaticResolverBundle\Models\Asset;
+
+use Exception;
+use Pimcore\Model\Asset;
+use Pimcore\Model\Asset\Listing;
+
+/**
+ * @internal
+ */
```
Comment (martineiber) [inline src/Models/Asset/AssetResolver.php:25]:
```suggestion
```

File: src/Models/Asset/AssetResolverInterface.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ */
+
+namespace Pimcore\Bundle\StaticResolverBundle\Models\Asset;
+
+use Pimcore\Model\Asset;
+use Pimcore\Model\Asset\Listing;
+
+/**
+ * @internal
+ */
```
Comment (martineiber) [inline src/Models/Asset/AssetResolverInterface.php]:
```suggestion
```

File: src/Models/Asset/AssetServiceResolver.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+namespace Pimcore\Bundle\StaticResolverBundle\Models\Asset;
+
+use Exception;
+use Pimcore\Model\Asset\Folder;
+use Pimcore\Model\Asset\Service;
+
+/**
+ * @internal
+ */
```
Comment (martineiber) [inline src/Models/Asset/AssetServiceResolver.php:25]:
```suggestion
```

File: src/Models/Asset/AssetServiceResolverInterface.php [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\StaticResolverBundle\Models\Asset;
+
+use Pimcore\Model\Asset\Folder;
+
+/**
+ * @internal
+ */
```
Comment (martineiber) [inline src/Models/Asset/AssetServiceResolverInterface.php]:
```suggestion
```

File: src/Models/DataObject/DataObjectResolver.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+namespace Pimcore\Bundle\StaticResolverBundle\Models\DataObject;
+
+use Exception;
+use Pimcore\Model\DataObject;
+use Pimcore\Model\DataObject\Listing;
+
+/**
+ * @internal
+ */
```
Comment (martineiber) [inline src/Models/DataObject/DataObjectResolver.php:25]:
```suggestion
```

File: src/Models/DataObject/DataObjectResolverInterface.php [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+ */
+
+namespace Pimcore\Bundle\StaticResolverBundle\Models\DataObject;
+
+use Pimcore\Model\DataObject;
+use Pimcore\Model\DataObject\Listing;
+
+/**
+ * @internal
+ */
```
Comment (martineiber) [inline src/Models/DataObject/DataObjectResolverInterface.php]:
```suggestion
```

File: src/Models/DataObject/DataObjectServiceResolver.php [hunk 7]
```diff
@@ -16,9 +16,14 @@
...
 namespace Pimcore\Bundle\StaticResolverBundle\Models\DataObject;
 
+use Exception;
+use Pimcore\Model\DataObject\Folder;
 use Pimcore\Model\DataObject\Service;
 
-class DataObjectServiceResolver implements DataObjectServiceResolverInterface
+/**
+ * @internal
+ */
```
Comment (martineiber) [inline src/Models/DataObject/DataObjectServiceResolver.php:25]:
```suggestion
```

File: src/Models/Document/DocumentServiceResolverInterface.php [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\StaticResolverBundle\Models\Document;
+
+use Pimcore\Model\Document\Folder;
+
+/**
+ * @internal
+ */
```
Comment (martineiber) [inline src/Models/Document/DocumentServiceResolverInterface.php]:
```suggestion
```

File: src/Models/Document/DocumentServiceResolver.php [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+namespace Pimcore\Bundle\StaticResolverBundle\Models\Document;
+
+use Exception;
+use Pimcore\Model\Document\Folder;
+use Pimcore\Model\Document\Service;
+
+/**
+ * @internal
+ */
```
Comment (martineiber) [inline src/Models/Document/DocumentServiceResolver.php:25]:
```suggestion

```

File: src/Models/Document/DocumentResolverInterface.php [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+ */
+
+namespace Pimcore\Bundle\StaticResolverBundle\Models\Document;
+
+use Pimcore\Model\Document;
+use Pimcore\Model\Document\Listing;
+
+/**
+ * @internal
+ */
```
Comment (martineiber) [inline src/Models/Document/DocumentResolverInterface.php]:
```suggestion
```

File: src/Models/Document/DocumentResolver.php [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+
+namespace Pimcore\Bundle\StaticResolverBundle\Models\Document;
+
+use Exception;
+use Pimcore\Model\Document;
+use Pimcore\Model\Document\Listing;
+
+/**
+ * @internal
+ */
```
Comment (martineiber) [inline src/Models/Document/DocumentResolver.php:25]:
```suggestion
```

File: src/Models/Asset/AssetResolverInterface.php [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+interface AssetResolverInterface
+{
+    public function getById(int|string $id, array $params = []): ?Asset;
```
Comment (martineiber) [inline src/Models/Asset/AssetResolverInterface.php:24]: We should but a space between each line. 
Not sure if CS-Fixer is checking that?

**Review by martineiber — CHANGES_REQUESTED** [review]
classes should not be internal.. otherwise we could not use them?

File: src/Models/Asset/AssetResolverInterface.php (hunk 12 above)
Comment (mattamon) [reply src/Models/Asset/AssetResolverInterface.php:24]: Done!

**Review by martineiber — APPROVED** [review]
LGTM!

**Pull request merged**